internal/providers/pke/pkeworkflow/generate_certificates_activity.go

```diff
@@ -16,51 +16,44 @@ package pkeworkflow
 
 import (
 	"context"
-	"fmt"
 
-	"github.com/banzaicloud/pipeline/pkg/auth"
+	"github.com/banzaicloud/pipeline/internal/cluster/clustersecret"
 	pkgSecret "github.com/banzaicloud/pipeline/pkg/secret"
-	"github.com/banzaicloud/pipeline/secret"
+	"github.com/pkg/errors"
 )
 
 const GenerateCertificatesActivityName = "pke-generate-certificates-activity"
 
 type GenerateCertificatesActivity struct {
-	clusters Clusters
+	secrets ClusterSecretStore
 }
 
-func NewGenerateCertificatesActivity(clusters Clusters) *GenerateCertificatesActivity {
+func NewGenerateCertificatesActivity(secrets ClusterSecretStore) *GenerateCertificatesActivity {
 	return &GenerateCertificatesActivity{
-		clusters: clusters,
+		secrets: secrets,
 	}
 }
 
+type ClusterSecretStore interface {
+	EnsureSecretExists(ctx context.Context, clusterID uint, secret clustersecret.NewSecret) (string, error)
```

Comment:
I think there should be a doc comment here too, stating what the method returns, because it is not obvious.

Reply:
Fair enough, I'll add better doc